client/settings/fraud-protection/advanced-settings/index.tsx

```diff
@@ -237,9 +245,51 @@ const FraudProtectionAdvancedSettingsPage: React.FC = () => {
 		};
 	}, [ isLoading ] );
 
+	const { isFRTReviewFeatureActive } = wcpaySettings;
+
+	const confirmLeaveCallback = useConfirmNavigation( () => {
+		const settingsChanged =
+			! isLoading &&
+			! isMatchWith(
+				readRuleset( advancedFraudProtectionSettings ),
+				protectionSettingsUI,
+				( source, target ) => {
+					for ( const rule in source ) {
+						// We need to skip checking the "block" property, as they are not the same with defaults.
+						if ( ! isFRTReviewFeatureActive && rule === 'block' ) {
+							continue;
+						}
+						if ( source[ rule ] !== target[ rule ] ) {
```

Comment:
Maybe instead of comparing the objects (they changed, so the reference isn't the same) we could compare strings with `JSON.stringify()`. I think it would be more resilient to updating the field and then changing it back to the original state.

Reply:
we need to skip some rules because in the UI the block setting is false, but on the default the block settings are true, and it creates a mismatch. This way we can skip properties that we don't care about not being same.